src/orquestra/sdk/mlflow/_connection_utils.py

```diff
@@ -5,11 +5,128 @@
 """Utilities for communicating with mlflow."""
 
 import os
+import warnings
 from pathlib import Path
+from typing import Optional, Tuple
 
+from requests import Response, Session
+
+from orquestra import sdk
+from orquestra.sdk._base import _env
 from orquestra.sdk._base._services import ORQUESTRA_BASE_PATH
+from orquestra.sdk.schema.configs import ConfigName
 
 DEFAULT_TEMP_ARTIFACTS_DIR: Path = ORQUESTRA_BASE_PATH / "mlflow" / "artifacts"
+RESOURCE_CATALOG_URI: str = "http://orquestra-resource-catalog.resource-catalog"
+
+
+# region: private
+def _is_executing_remoteley() -> bool:
+    """
+    Determine whether the code is being executed locally, or on a cluster/studio.
+    """
+    # TODO: at present this uses the platform-specific environment variables, assuming
+    # that if all of them are set then we must be executing remotely. This assumption
+    # is not 100% - power users may set the envvars locally for fine-grained control.
+    # However, the platform will shortly be adding an envvar specifically to track
+    # whether we're executing remotely. Once that is done, this function should be
+    # updated to check that specifically. Tests should be added for this function at
+    # that time.
+    envvars = [
+        _env.PASSPORT_FILE_ENV,
+        _env.MLFLOW_CR_NAME,
+        _env.MLFLOW_PORT,
+        _env.MLFLOW_ARTIFACTS_DIR,
+    ]
+    if None in [os.getenv(envvar) for envvar in envvars]:
+        return False
+    return True
+
+
+def _get_mlflow_cr_name_and_port() -> Tuple[str, str]:
+    """
+    Reads in the MLFlow custom resource name and port from environment variables.
+
+    The environment variables are:
+    - ORQ_MLFLOW_CR_NAME
+    - ORQ_MLFLOW_PORT
+
+    Example usage:
+
+    ```python
+    mlflow_cr_name, mlflow_port = _get_mlflow_cr_name_and_port()
+    ```
+
+    Raises:
+        EnvironmentError: when either of the environment variables are not set.
+    """
+
+    assert (
+        _is_executing_remoteley()
+    ), "This function should not be called when running locally."
+
+    if not (cr_name := os.getenv(_env.MLFLOW_CR_NAME)):
+        raise EnvironmentError(
+            f"The '{_env.MLFLOW_CR_NAME}' environment variable is not set."
+        )
+    if not (port := os.getenv(_env.MLFLOW_PORT)):
+        raise EnvironmentError(
+            f"The '{_env.MLFLOW_PORT}' environment variable is not set."
+        )
+
+    return cr_name, port
+
+
+def _read_passport_token() -> str:
+    """
+    Reads in the token.
+
+    The file path is specified by the PASSPOT_FILE_ENV environment variable.
+
+    Raises:
+        EnvironmentError: when the PASSPORT_FILE_ENV environment variable is not set.
+    """
+    if not (passport_file_path := os.getenv(_env.PASSPORT_FILE_ENV)):
+        raise EnvironmentError(
+            f"The '{_env.PASSPORT_FILE_ENV}' environment variable is not set."
+        )
+    return Path(passport_file_path).read_text()
+
+
+def _make_workspace_zri(workspace_id: str) -> str:
```

Comment:
Suggestion: Im pretty sure we have this logic somewhere in the code already. Could we reuse it?

Reply:
Good idea, I've moved it to the spaces api.